coaster/sqlalchemy/statemanager.py

```diff
@@ -526,9 +546,13 @@ def __call__(self, *args, **kwargs):
         # Call the transition function
         try:
             result = self.statetransition.func(self.obj, *args, **kwargs)
+        except AbortTransition as e:
+            transition_exception.send(self.obj, transition=self.statetransition, exception=e)
+            return None if len(e.args) == 0 else e.args[0] if len(e.args) == 1 else e.args
```

Comment:
This is unsafe. You must insist on `AbortTransition` accepting a single parameter, and returning it as is.